packages/volto/src/components/manage/Widgets/SchemaWidget.jsx

```diff
@@ -142,160 +171,597 @@ const makeFieldsetList = (listOfFieldsets, intl) => {
+  method: (intl) => ({
+    type: 'string',
+    widget: 'textarea',
+    factory: 'textarea',
+  }),
+});
+
+/**
+ * schemaField used for modal form, when editing a field
+ * - based on the factory a set of fields is presented
+ * - fields can be moved to another fieldset
+ * @param {string} factory - the kind of field
+ * @param {Object} intl
+ * @param {*} fieldsets
+ * @param {Boolean} allowEditId
+ * @param {Object} extraFields
+ * @return {Object} - schema
+ */
+const schemaField = (
+  factory,
+  intl,
+  fieldsets,
+  allowEditId,
+  extraFields = {},
+) => {
+  const utility = config.getUtility({
+    name: factory,
+    type: 'fieldFactoryProperties',
+  });
+
+  const properties = utility.method
+    ? utility.method(intl)
+    : {
+        minLength: {
+          type: 'integer',
+          title: intl.formatMessage(messages.minLength),
+        },
+        maxLength: {
+          type: 'integer',
+          title: intl.formatMessage(messages.maxLength),
+        },
+        default: {
+          type: 'string',
+          title: intl.formatMessage(messages.defaultValue),
+        },
+        placeholder: {
+          type: 'string',
+          title: intl.formatMessage(messages.placeholder),
+        },
+      };
+
+  return {
+    fieldsets: [
+      {
+        id: 'default',
+        title: 'default',
+        fields: [
+          ...keys(extraFields),
+          ...(allowEditId ? ['id'] : []),
+          ...['title', 'description', 'parentFieldSet', 'queryParameterName'],
+          ...keys(properties),
+          ...['required'],
+        ],
+      },
+    ],
+    properties: {
+      ...extraFields,
+      ...(allowEditId
+        ? {
+            id: {
               type: 'string',
-              title: intl.formatMessage(messages.choices),
-              widget: 'textarea',
+              title: intl.formatMessage(messages.idTitle),
             },
-          };
-        default:
-          return {
-            minLength: {
-              type: 'integer',
-              title: intl.formatMessage(messages.minLength),
-            },
-            maxLength: {
-              type: 'integer',
-              title: intl.formatMessage(messages.maxLength),
-            },
-          };
-      }
-    })(factory),
-  },
-  required: ['type', 'title'],
-});
+          }
+        : {}),
+      title: {
+        type: 'string',
+        title: intl.formatMessage(messages.title),
+      },
+      description: {
+        type: 'string',
+        widget: 'textarea',
+        title: intl.formatMessage(messages.description),
+      },
+      parentFieldSet: {
+        type: 'string',
+        title: intl.formatMessage(messages.parentFieldSet),
+        choices: makeFieldsetList(fieldsets),
+      },
+      queryParameterName: {
+        type: 'string',
+        title: intl.formatMessage(messages.queryParameterName),
+        description: intl.formatMessage(messages.queryParameterNameDescription),
+      },
```

Comment:
I tried adding a queryParameterName to a field in a content type schema, but it doesn't do anything.

Reply:
If it doesn't do anything, it's bad UX to show it when editing a content type schema in the schemaeditor.

Reply:
I disagree, the schema widget is not tied to a specific form library and set of widgets. You can decide to use your own set of widgets which do make use of these values. This is not something new in Plone, there are plenty of fields on a content type or settings in the control panels which are not used by stock plone or shown in the ui but can be used if somebody wants that.

Reply:
Both query param and placeholder are now opt-in.